### PR TITLE
Improve cache key generation in WP_Comment_Query

### PR DESCRIPTION
**Pull request by uday-kokitkar**
Trac ticket: https://core.trac.wordpress.org/ticket/55460
XWP: https://xwp-co.atlassian.net/browse/INITS-502

---
**This Pull Request is for code review only. Please keep all other discussion in the Trac ticket. Do not merge this Pull Request. See [GitHub Pull Requests for Code Review](https://make.wordpress.org/core/handbook/contribute/git/github-pull-requests-for-code-review/) in the Core Handbook for more details.**

### CODE REVIEW

File: src/wp-includes/class-wp-comment-query.php [hunk 1]
```diff
@@ -444,10 +444,12 @@ public function get_comments() {
 
 		/*
 		 * Only use the args defined in the query_var_defaults to compute the key,
-		 * but ignore 'fields', which does not affect query results.
+		 * but ignore 'fields', 'update_comment_meta_cache', 'update_comment_post_cache' which does not affect query results.
 		 */
 		$_args = wp_array_slice_assoc( $this->query_vars, array_keys( $this->query_var_defaults ) );
 		unset( $_args['fields'] );
+		unset( $_args['update_comment_meta_cache'] );
```
Comment (manuelRod) [inline src/wp-includes/class-wp-comment-query.php]: unset( $_args['fields'], $_args['update_comment_meta_cache'], ... )

File: tests/phpunit/tests/comment/query.php [hunk 2]
```diff
@@ -4941,4 +4941,53 @@ public static function filter_comments_pre_query_and_set_comments( $comments, $q
...
+		$q1->query( $_args );
+
+		$num_queries_all_args = $wpdb->num_queries;
+
+		// Unset 'fields'.
+		unset( $_args['fields'] );
+		$q2 = new WP_Comment_Query();
+		$q2->query( $_args );
+
+		$num_queries_skip_fields = $wpdb->num_queries;
```
Comment (manuelRod) [inline tests/phpunit/tests/comment/query.php]: you can use get_num_queries() instead, and avoid global $wpdb

**Review by manuelRod — CHANGES_REQUESTED** [review]
I left few nitpicks

File: tests/phpunit/tests/comment/query.php (hunk 2 above)
Comment (uday-kokitkar) [reply tests/phpunit/tests/comment/query.php]: That's a good alternative. I followed code from other tests.

File: src/wp-includes/class-wp-comment-query.php (hunk 1 above)
Comment (uday-kokitkar) [reply src/wp-includes/class-wp-comment-query.php]: Done.

**Review by manuelRod — APPROVED** [review]

**Pull request closed without merging**